src/hooks/useTxStatus/useTxStatus.ts

```diff
@@ -0,0 +1,71 @@
+import { usePrevious } from '@chakra-ui/react'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useEffect, useMemo } from 'react'
+import { selectTxById } from 'state/slices/selectors'
+import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
+import { useAppSelector } from 'state/store'
+
+type UseTxStatusProps = {
+  accountId?: AccountId | null
+  txId?: string | null
+  onTxStatusConfirmed?: () => Promise<void>
+  onTxStatusPending?: () => Promise<void>
+  onTxStatusFailed?: () => Promise<void>
+  onTxStatusChanged?: (status: TxStatus) => Promise<void>
```

Comment:
suggestion: remove me, this isn't consumed at the moment, and the 3 handlers above would handle this

Reply:
fixed in [77b81a5](https://github.com/shapeshift/web/pull/7043/commits/77b81a593dc0a6c7088883c3ee9d54c59b297636)